app/models/event.rb

```diff
@@ -20,18 +20,52 @@ class Event < ActiveRecord::Base
 
   accepts_nested_attributes_for :users
   accepts_nested_attributes_for :user_events
-  validates_presence_of :subject
+  validates_presence_of :subject, :datetime, :location
+  validate :datetime_no_less_than_15_mins_before_now
 
-  after_create :send_email
-  after_update :send_email
+  after_create :send_default_email
+  after_update :update_db, :send_updated_email
 
-  def send_email
+  DELAY = 3.hours.from_now
```

Comment:
because it's a piece of executable code. and constant will be initialised when an instance of class is initialised. So 3.hours.from_now can be quite not 3 hours from now. Simple rule - constant for simple values.